ls1intum/Artemis
# `Development`: Update dependencies for Java Blackbox exercise template

### Checklist
#### General



#### Changes affecting Programming Exercises
- [ ] ~~I tested **all** changes and their related features with **all** corresponding user types on Test Server 1 (Atlassian Suite).~~ Not applicable.
<br><br>
#### `Note`: The pipeline-helper update is included in the new docker image for blackbox/DejaGnu project types
### Motivation and Context


The pipeline-helper is a tool that among other things performs multiple checks on the code of blackbox exercises. Java 21 is going to introduces new ways to formulate main methods (https://openjdk.org/jeps/445), which could potentially break the previous implementation of the main method finder utility of the pipeline helper. This PR serves to counteract that by adapting the pipeline-helper accordingly.

### Description

Aside from the more readable commands, the main method checker now uses reflection to find main methods in the specified directory (here: `target/classes`), so it looks through all the compiled classes of the repository.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course
- TS2 (Jenkins Setup)

1. Log in to Artemis
2. Navigate to Course Management
3. Create a Java programming exercise with online editor enabled, and choose the new `DejaGnu` project type.
4. Make sure the BASE build fails, and the SOLUTION build works.
5. When clicking on the SOLUTION build result, it should display 6 passing test cases.

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
